lib/chef/provider.rb

```diff
@@ -211,10 +211,29 @@ def self.include_resource_dsl_module(resource)
           extend Forwardable
           define_singleton_method(:to_s) { "#{resource_class} forwarder module" }
           define_singleton_method(:inspect) { to_s }
+          # Add a delegator for each explicit property that will get the *current* value
+          # of the property by default instead of the *actual* value.
+          resource.class.properties.each do |name, property|
+            class_eval(<<-EOM, __FILE__, __LINE__)
+              def #{name}(*args, &block)
```

Comment:
Yeah, I can add a rescue to fall back on define_method in that case. We do that in the `property` declaration spot.